src/main/java/net/dv8tion/jda/api/events/Event.java

```diff
@@ -45,6 +49,7 @@ public Event(@Nonnull JDA api, long responseNumber)
     {
         this.api = api;
         this.responseNumber = responseNumber;
+        this.rawData = api instanceof JDAImpl && ((JDAImpl) api).isEventPassthrough() ? SocketHandler.CURRENT_EVENT.get() : null;
```

Comment:
You don't need this instanceof check. JDA will always be JDAImpl

Reply:
This is for the purpose of mocking.

Reply:
We do this cast blindly throughout all of JDA. What makes you think this situation is different?